workflow-core/src/commonMain/kotlin/com/squareup/workflow1/SessionWorkflow.kt

```diff
@@ -0,0 +1,154 @@
+package com.squareup.workflow1
+
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * @see [StatefulWorkflow] for the main description of this class.
+ *
+ * This is an extension of the API that allows for [initialState] to also receive a [CoroutineScope]
+ * that corresponds with the lifetime of this Workflow. In other words, the [CoroutineScope] is
+ * created when an session of this Workflow is active. A session is active from the first time one
+ * node of it is rendered, until it is no longer rendered.
+ *
+ * This API extension exists on [StatefulWorkflow] as well, but it is confusing because the version
+ * of [initialState] that does not have the [CoroutineScope] must also be implemented as it is
+ * an abstract fun, even though it would never be used.
+ * With this version, that confusion is removed and only the version of [initialState] with the
+ * [CoroutineScope] must be implemented.
+ */
+@WorkflowExperimentalApi
+public abstract class SessionWorkflow<
+  in PropsT,
+  StateT,
+  out OutputT,
+  out RenderingT
+  > : StatefulWorkflow<PropsT, StateT, OutputT, RenderingT>() {
+
+  /**
+   * @see [StatefulWorkflow.initialState] for kdoc on the base function of this method.
+   *
+   * This version adds the following:
+   * @param workflowScope A [CoroutineScope] that has been created when this Workflow is first
+   * rendered and canceled when it is no longer rendered.
+   *
+   * This [CoroutineScope] be used to own the transforms on, e.g. a [StateFlow][kotlinx.coroutines.flow.StateFlow],
+   * in order to link them to the lifetime of this Workflow's session - from first rendering until
+   * no longer rendered. e.g.,
+   * ```
+   * val transformedStateFlow = myStateFlow1.combine(myStateFlow2).
+   *   stateIn(workflowScope, SharingStarted.Eagerly, myStateFlow1.value + myStateFlow2.value)
+   * ```
```

Comment:
We usually indent code blocks instead of relying on triple backtick. Same formatting. 

```suggestion
   * This [CoroutineScope] be used to:
   *
   *  - build reliable set up and teardown hooks, e.g. via
   *    [Job.disposeOnCompletion][kotlinx.coroutines.disposeOnCompletion]
   *
   *  -  own the transforms on a [StateFlow][kotlinx.coroutines.flow.StateFlow],
   *     linking them to the lifetime of a Workflow's session. For example,
   *     here is how you might safely combine two `StateFlow`s:
   *
   *     data class MyState(
   *       val derivedValue: String,
   *       val derivedWorker: Worker<String>
   *     )
   *
   *     override fun initialState(
   *       props: Unit,
   *       snapshot: Snapshot?,
   *       workflowScope: CoroutineScope
   *     ): MyState {
   *       val transformedStateFlow = stateFlow1.combine(stateFlow2).
   *         stateIn(workflowScope, SharingStarted.Eagerly, "${stateFlow1.value}-${stateFlow2.value}")
   *
   *       return MyState(
   *         transformedStateFlow.value,
   *         transformedStateFlow.asWorker()
   *       )
   *     }
```

Reply:
Changed manually.